Enable click handling, as well as hover, on QuestionButton

azangru opened the pull request:
## Type
- Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1343

## Description
- Add click handling to `QuestionButton`. A click on `QuestionButton` will show the tooltip; the next click will hide it, and so on.
- Also, keep the previous behaviour on hover (wait for a while and show the tooltip)
- Update some CSS to prevent text selection and other nuisance on repeat clicks

## Deployment URL
http://question-button.review.ensembl.org

imransl commented on src/ensembl/src/shared/components/question-button/QuestionButton.tsx:
```diff
@@ -14,10 +14,10 @@
 import classNames from 'classnames';
 
-import useHover from 'src/shared/hooks/useHover';
+import { useQuestionButtonBehavior } from './useQuestionButtonBehavior';
```
Aren't we sticking with UK spellings for variables?

jyothishnt replied:
we may need to change this too https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item/ProteinsListItem.tsx#L102 😆

jyothishnt replied:
I just did a quick check in the code and found we are sticking on to British English. Sorry @azangru  :-)

azangru replied:
We can't change that, Jyo; it's the browser API, and browsers speak American English:

https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView

This is why I would love us to use American English in code. All the other examples of the word `behaviour` in our code are for display labels; which is totally fine, we could write labels in Chinese hyerogliphs for all we care. But the code itself would be much more consistent if it stuck to American spelling 😞

jyothishnt replied:
oh oh!

azangru replied:
It is with heavy heart that I have changed behavior for behaviour 😞 

✅

imransl commented on src/ensembl/src/shared/components/question-button/useQuestionButtonBehavior.tsx:
```diff
@@ -0,0 +1,109 @@
+ */
+
+import { useReducer, useEffect } from 'react';
+import useHover from 'src/shared/hooks/useHover';
```
Separate these imports?

jyothishnt commented:
When I click to show tooltip, should it close only when I click outside?
It closes when I hover the icon for few seconds and move away. Just checking if this is the correct behaviour.

jyothishnt commented on src/ensembl/src/shared/components/question-button/QuestionButton.test.tsx:
```diff
@@ -0,0 +1,128 @@
+      expect(questionButton.querySelector('.pointerBox')).toBeFalsy();
+
+      act(() => {
+        jest.advanceTimersByTime(TOOLTIP_TIMEOUT + 10);
```
TOOLTIP_TIMEOUT = 800  Is it in milliseconds? If so, does it wait for 8 seconds here when we run this test?

azangru commented:
> Currently if the tooltip is activated by click and we do a hover action (hover in and out), it closes the tooltip.

Thanks for spotting this, @jyothishnt . I updated the code to prevent this from happening, and added a test to confirm that this is now the case.
